### ISSUE TITLE
Further improvements of the usage stats log file processing

### ISSUE THREAD

**Issue by bozana** [pkp/pkp-lib#9708]
1. Consider these two cases regarding the jobs:

- If the re-processing of the same log file has started next day: This case can only occur if the admin explicitly does it. But this we could eventually solve by having an ID, to avoid them from being duplicated on the queue. I will need to think about how to solve it. Maybe in a new issue. It is probably also a rare case, because I hope that admin will not do such a think :-)
- If log files processing (e.g. when the admin re-process the whole month) takes longer than a day, and the normal scheduled task that runs every day (and that would process the log file from yesterday) is run on the next day: Then the new scheduled task would create a chain of the new jobs (for the new log file) and those can be run in parallel with the jobs from the re-processing the whole month. This would again lead to enormous DB tables that would break the processing. This I do not know how to solve, I would need to investigate what is possible. But also this I would like to leave for a new issue.

2. Use FilesystemIterator instead of opendir() in the FileLoader::claimNextFile()
3. Do we need to log the error when the job in a chain fails in PKPUsageStatsLoader: 
`Bus::chain($jobs)->catch(function (Throwable $e) { // error_log...`
